tidy3d/components/simulation.py

```diff
@@ -825,17 +825,27 @@ def tfsf_with_symmetry(cls, val, values):
 
     @pydantic.validator("boundary_spec", always=True)
     def boundaries_for_zero_dims(cls, val, values):
-        """Error if an absorbing boundary is used along a zero dimension."""
+        """Error if an absorbing boundary or symmetry is used along a zero dimension."""
         boundaries = val.to_list
         size = values.get("size")
-        for dim, (boundary, size_dim) in enumerate(zip(boundaries, size)):
+        symmetry = values.get("symmetry")
+        for dim, (boundary, symmetry_dim, size_dim) in enumerate(zip(boundaries, symmetry, size)):
             num_absorbing_bdries = sum(isinstance(bnd, AbsorberSpec) for bnd in boundary)
             if num_absorbing_bdries > 0 and size_dim == 0:
                 raise SetupError(
                     f"The simulation has zero size along the {'xyz'[dim]} axis, so "
                     "using a PML or absorbing boundary along that axis is incorrect. "
                     f"Use either 'Periodic' or 'BlochBoundary' along {'xyz'[dim]}."
                 )
+
+            if symmetry_dim != 0 and size_dim == 0:
+                log.warning(
+                    f"The simulation has zero size along the {'xyz'[dim]} axis, so "
+                    "using symmetry along that axis is likely not necessary and "
+                    "could potentially lead to incorrect results. "
+                    f"Consider setting symmetry to `0` along {'xyz'[dim]}."
```

Comment:
`symmetry` -> `'Simulation.symmetry'` might make it more clear which field specifically to set.